push_notifications/wns.py

```diff
@@ -0,0 +1,359 @@
+"""
+Windows Notification Service
+
+Documentation is available on the Windows Dev Center:
+https://msdn.microsoft.com/en-us/windows/uwp/controls-and-patterns/
+tiles-and-notifications-windows-push-notification-services--wns--overview
+"""
+
+import json
+import xml.etree.ElementTree as ET
+from xml.etree.ElementTree import tostring
+
+try:
+	from urllib.error import HTTPError
+	from urllib.request import Request, urlopen
+except ImportError:
+	# Python 2 support
+	from urllib2 import HTTPError, Request, urlopen
+
+from django.core.exceptions import ImproperlyConfigured
+
+from . import NotificationError
+from .settings import PUSH_NOTIFICATIONS_SETTINGS as SETTINGS
+
+WNS_ACCESS_URL = "https://login.live.com/accesstoken.srf"
```

Comment:
It's useful if the upstream URL changes and the DPN project is unmaintained at that time, and/or someone runs a self-hosted WNS dispatch server at a different address.

Both are unlikely, but it mirrors the behaviour in GCM/APNS.